### DISCUSSION TITLE
Add esm target to @directus/app package

### DISCUSSION THREAD

**Discussion by B-Esmaili** [directus/directus#14921] [Feature Requests]
As most of app related components are in `@directus/app` package and there is no way to reference those components from extensions , internal components are not usable in extensions packages. It would be great help for extension developers to be able to reference internal components from extensions.Right now i can install `@directus/app` package but am unable to import from them because they are all production bundles. I think adding `esm` target to this package would make it totally usable in extensions.

**Comment by rijkvanzanten** [comment]
The `@directus/app` package only contains the bundled output of the app, not the individual components/source code. That being said, every base component (`v-icon` etc) is already globally registered, and can be used in any extension without importing it 👍🏻

**Reply by B-Esmaili** [reply]
Yes, I am aware of  `v-*` components but some composables like `useItem` should have been included in `@directus/shared` but are instead inside `@directus/app`. It seems to me composables like this one should be accessible from extensions just like `useItems` does.

**Reply by rijkvanzanten** [reply]
The difficulty here is that exposing parts like that means that every change to the part is a breaking change for extension builders, which makes maintenance quite a bit harder. That's why we're only exposing certain things, and not everything at once.

(Also we don't really want to ship the full source of the app in the bundle, as that would greatly increase the install size of the platform)

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
